neqo-transport/src/cid.rs

```diff
@@ -405,7 +405,7 @@ impl ConnectionIdStore<[u8; 16]> {
     pub fn retire_prior_to(&mut self, retire_prior: u64) -> Vec<u64> {
         let mut retired = Vec::new();
         self.cids.retain(|e| {
```

Comment:
Nit: could as well use `SmallVec::drain_filter`. Feel free to ignore.

https://docs.rs/smallvec/latest/smallvec/struct.SmallVec.html#method.drain_filter

Reply:
That seems to be behind a feature and is unstable? https://docs.rs/smallvec/latest/smallvec/#drain_filter

Can obviously add that feature. Are there others we should add? `union`, `const_generics` and `const_new` seem useful?